chiselFrontend/src/main/scala/Chisel/When.scala

```diff
@@ -2,9 +2,12 @@
 
 package Chisel
 
+import scala.language.experimental.macros
+
 import internal._
 import internal.Builder.pushCommand
 import internal.firrtl._
+import internal.sourceinfo.{SourceInfo, WhenTransform}
 
 object when {  // scalastyle:ignore object.name
```

Comment:
Finalize WhenContext. Given how low-level it acts (pushing commands and very precisely deferring executing certain commands), I have problems envisioning how it would be properly extended versus someone just writing their own variant.

I don't understand the desire to overcomplicate every part of the Chisel API in this PR. This also really muddles the core reason for all these macro transformations: to allow direct applies on Chisel.Data.

Macros are complex (and slow compilation). Should be used only when necessary.